### ISSUE TITLE
sys.exit called on rst parse errors

### ISSUE THREAD

**Issue by valhallasw** [getpelican/pelican#1759]
The docutils.core.Publisher backend is called in a such a way that it doesn't raise an Exception, but calls sys.exit instead. This makes it harder to use pelican as library.

How to reproduce this?

Using an rst file with invalid syntax, run:

```
from pelican import Pelican
from pelican.settings import read_settings

p = Pelican(read_settings('pelicanconf.py'))
try:
    p.run()
except Exception as e:
   print "Caught exception: ", e

print "Continue after failure"
```

Expected behavior:

```
$ python test.py
....../file.rst:29: (ERROR/3) Unknown directive type "fdsaf".

<details>

Caught exception: SomeException
Continue after failure
```

Actual behavior:

```
$ python test.py
....../file.rst:29: (ERROR/3) Unknown directive type "fdsaf".

<details>

$ (silently exits)
```

This is caused because of  `enable_exit_status=True` which is passed to docutils.core.Publisher.publish since #1224 was merged.

Unfortunately, I don't know enough about the internal structure of Pelican to suggest where the docutils Exception should be changed into a sys.exit, but I'm inclined to say this should be close to the entry point. This allows people who call Pelican as a library to catch an Exception as-is, instead of resorting to the workaround

```
try:
    p.run()
except SystemExit as e:
   print "Caught sys.exit."
```

**Comment by howthebodyworks** [comment]
Moreover, it makes hard to work out where exceptions are raised, since the original traceback is lost.

**Comment by avaris** [comment]
I'm not sure why this was considered a good idea in the first place. Exiting the process for an error in a single `.rst` file does sound like the definition of overkill. Instead it should log the error and skip the file.

**Comment by omiday** [comment]
I was going to submit a PR but first need to fix #2032.

**Comment by leotrs** [comment]
@omiday did that PR ever get done?

**Comment by justinmayer** [comment]
Status: PR #2047 still awaiting review.

**Comment by justinmayer** [comment]
Addressed by #2438.
